### PR TITLE
Rework Trainer building in ML CLI

### PR DESCRIPTION
**Pull request by kaminow**
There were a couple of things that didn't work the way I wanted them to in the ML CLI that are fixed in this PR:
* Passing a `Trainer` config file ignore all other CLI args -> args from  config file are loaded, and then any CLI args that are specified override those in the file
* Continuing an existing run didn't do anything -> now works as expected

### CODE REVIEW

File: asapdiscovery-ml/asapdiscovery/ml/cli.py [hunk 1]
```diff
@@ -1392,151 +1462,165 @@ def build_and_train_schnet(
...
+                    {
+                        k: v
+                        for k, v in trainer_kwargs[config_name].items()
+                        if v is not None
+                    }
+                )
+            else:
+                config_trainer_kwargs[config_name] = trainer_kwargs[config_name]
+
+        trainer_kwargs = config_trainer_kwargs
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/cli.py:1600]: This has been repeated a lot, you can just make it a function that lives in this module so you only have to update one spot

File: asapdiscovery-ml/asapdiscovery/ml/cli.py [hunk 2]
```diff
@@ -1392,151 +1462,165 @@ def build_and_train_schnet(
...
-        # Save Trainer
-        if trainer_config_cache and (
-            (not trainer_config_cache.exists()) or overwrite_trainer_config_cache
-        ):
-            trainer_config_cache.write_text(t.json())
+    # Save Trainer
+    if trainer_config_cache and (
+        (not trainer_config_cache.exists()) or overwrite_trainer_config_cache
+    ):
+        trainer_config_cache.write_text(t.json())
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/cli.py:1623]: Same here, just mash into a function that handles this (if possible)

**Review by hmacdope — CHANGES_REQUESTED** [review]
Just make some utility functions to avoid some repetition, other than that looks good.

**Comment by kaminow** [comment]
@hmacdope good call, should be good to go now

**Review by hmacdope — APPROVED** [review]
LGTM, thanks for making the change.

**Comment by hmacdope** [comment]
Ah @kaminow will need to fix conflicts

**Comment by kaminow** [comment]
oop yeah I will fix

**Pull request merged**